src/components/MainTable/MainTable.tsx

```diff
@@ -316,6 +307,27 @@ const MainTable = ({
     onUpdateSort && onUpdateSort(newSort);
   };
 
+  const sortedRows = useMemo(
+    () =>
+      sortRows({
+        currentSortDirection,
+        currentSortKey,
+        rows,
+        sortable,
+        sortFunction,
+      }),
+    [currentSortDirection, currentSortKey, rows, sortable, sortFunction]
+  );
+
+  const {
+    pageData: finalRows,
+    currentPage,
+    paginate: setCurrentPage,
+  } = usePagination(sortedRows, {
+    itemsPerPage: paginate,
+    autoResetPage: true,
```

Comment:
Should this be exposed as a component prop?

Reply:
Not at this stage, I don't think. We can always add this if needed.